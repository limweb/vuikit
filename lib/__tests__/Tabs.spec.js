import { createVue, destroyVM, queryByTag, queryByTagAll, toJSON } from './_util'
import waitForUpdate from './_wait-for-update'

describe.skip('Tabs', () => {
  let vm
  afterEach(() => {
    destroyVM(vm)
  })

  it('renders correctly', () => {
    vm = createVue(`
      <vk-tabs :activeTab="1" bottom center flip width="1-1">
        <vk-tab label="Tab 1">Content Tab 1</vk-tab>
        <vk-tab label="Tab 2" disabled>Content Tab 2</vk-tab>
        <vk-tab>
          <template slot="header">Header from slot</template>
        </vk-tab>
      </vk-tabs>
    `)
    expect(toJSON(vm)).toMatchSnapshot()
  })

  it('renders correctly verticaly', () => {
    vm = createVue(`
      <vk-tabs-vertical :activeTab="1" flip width="1-1" contentWidth="1-2">
        <vk-tab>Content Tab 1</vk-tab>
      </vk-tabs-vertical>
    `)
    expect(toJSON(vm)).toMatchSnapshot()
  })

  it('triggers event change when a tab is clicked', done => {
    vm = createVue(`
      <vk-tabs activeTab="tab1">
        <vk-tab alias="tab1">Content Tab 1</vk-tab>
        <vk-tab alias="tab2">Content Tab 2</vk-tab>
      </vk-tabs>
    `)

    const cb = jest.fn()
    const tabsWrapper = queryByTag(vm, 'vk-tabs')
    const tabs = queryByTagAll(vm, 'vk-tab')

    tabsWrapper.$on('change', cb)
    waitForUpdate(() => {
      tabs[1].$el.children[0].click()
    }).then(() => {
      const args = cb.mock.calls[0]
      expect(cb).toHaveBeenCalled()
      expect(args).toHaveLength(1)
      expect(args[0]).toEqual('tab2')
    }).then(done)
  })
})
