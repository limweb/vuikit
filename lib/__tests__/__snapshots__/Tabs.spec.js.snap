exports[`Tabs renders correctly 1`] = `
Object {
  "child": Array [
    Object {
      "attr": Object {
        "class": Array [
          "uk-flex",
          "uk-flex-column-reverse",
        ],
      },
      "child": Array [
        Object {
          "attr": Object {
            "class": Array [
              "uk-tab-center",
              "uk-tab-center-bottom",
            ],
          },
          "child": Array [
            Object {
              "attr": Object {
                "class": Array [
                  "uk-tab",
                  "uk-tab-grid",
                  "uk-tab-flip",
                  "uk-tab-bottom",
                ],
              },
              "child": Array [
                Object {
                  "attr": Object {
                    "class": Array [
                      "uk-width-1-1",
                      "uk-active",
                    ],
                  },
                  "child": Array [
                    Object {
                      "child": Array [
                        Object {
                          "node": "text",
                          "text": "Tab1",
                        },
                      ],
                      "node": "element",
                      "tag": "a",
                    },
                  ],
                  "node": "element",
                  "tag": "li",
                },
                Object {
                  "node": "text",
                  "text": "",
                },
                Object {
                  "attr": Object {
                    "class": Array [
                      "uk-width-1-1",
                      "uk-disabled",
                    ],
                  },
                  "child": Array [
                    Object {
                      "child": Array [
                        Object {
                          "node": "text",
                          "text": "Tab2",
                        },
                      ],
                      "node": "element",
                      "tag": "a",
                    },
                  ],
                  "node": "element",
                  "tag": "li",
                },
                Object {
                  "node": "text",
                  "text": "",
                },
                Object {
                  "attr": Object {
                    "class": "uk-width-1-1",
                  },
                  "child": Array [
                    Object {
                      "child": Array [
                        Object {
                          "node": "text",
                          "text": "Headerfromslot",
                        },
                      ],
                      "node": "element",
                      "tag": "a",
                    },
                  ],
                  "node": "element",
                  "tag": "li",
                },
              ],
              "node": "element",
              "tag": "ul",
            },
          ],
          "node": "element",
          "tag": "div",
        },
        Object {
          "node": "text",
          "text": "",
        },
        Object {
          "attr": Object {
            "class": "uk-margin",
          },
          "node": "element",
          "tag": "div",
        },
      ],
      "node": "element",
      "tag": "div",
    },
  ],
  "node": "root",
}
`;

exports[`Tabs renders correctly verticaly 1`] = `
Object {
  "child": Array [
    Object {
      "attr": Object {
        "class": Array [
          "uk-grid",
          "uk-flex",
          "uk-flex-row-reverse",
        ],
      },
      "child": Array [
        Object {
          "attr": Object {
            "class": "uk-width-medium-1-1",
          },
          "child": Array [
            Object {
              "attr": Object {
                "class": Array [
                  "uk-tab",
                  "uk-tab-right",
                ],
              },
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "uk-active",
                  },
                  "child": Array [
                    Object {
                      "child": Array [
                        Object {
                          "node": "text",
                          "text": "",
                        },
                      ],
                      "node": "element",
                      "tag": "a",
                    },
                  ],
                  "node": "element",
                  "tag": "li",
                },
              ],
              "node": "element",
              "tag": "ul",
            },
          ],
          "node": "element",
          "tag": "div",
        },
        Object {
          "node": "text",
          "text": "",
        },
        Object {
          "attr": Object {
            "class": "uk-width-medium-1-2",
          },
          "child": Array [
            Object {
              "node": "element",
              "tag": "div",
            },
          ],
          "node": "element",
          "tag": "div",
        },
      ],
      "node": "element",
      "tag": "div",
    },
  ],
  "node": "root",
}
`;
